tox.ini

```diff
@@ -3,29 +3,45 @@ max-line-length = 120
 
 [tox]
 skipsdist = True
-envlist = lint,py3
+envlist = lint,unit
 
 [testenv]
 basepython = python3
 setenv =
     PYTHONPATH={toxinidir}:{toxinidir}/lib
+
+[testenv:format]
+deps =
+    black
+commands =
+    black {toxinidir}/reactive {toxinidir}/lib {toxinidir}/tests/integration {toxinidir}/tests/unit
+
+[testenv:unit]
 deps =
     jinja2
     netaddr<=0.7.19
     pyyaml
     pytest
     pytest-mock
     pytest-cov
-    black
-    flake8
     charms.unit_test
 commands =
     pytest --cov-report term-missing \
         --cov lib --cov-fail-under 33 \
-        --tb native -s {posargs}
+        --tb native -s {posargs} {toxinidir}/tests/unit
+
+[testenv:integration]
+deps =
+    pytest
+    pytest-operator
+    charms.unit_test
+commands = pytest --tb native --show-capture=no --disable-pytest-warnings --log-cli-level=INFO -s {posargs} {toxinidir}/tests/integration
```

Comment:
OH!  this needs to be added to the .github actions. 

see 
https://github.com/charmed-kubernetes/charm-kubernetes-control-plane/blob/main/.github/workflows/main.yml#L34-L53